include/aspect/utilities.h

```diff
@@ -813,6 +813,42 @@ namespace aspect
          */
         std_cxx11::unique_ptr<aspect::Utilities::AsciiDataLookup<1> > lookup;
     };
+
+    /**
+     * A weighted p norm average caluclator:
+     * $\text{weighed\_p\_norm\_average} =
+     * \frac{1}{k}(\sum_{i=1}^k \big(\text{weights}_i \text{values}_i^p)\big)^{\frac{1}{p}}$.
+     *
+     * There have been several special cases implemented, mostly for performance reasons, and
```

Comment:
so in the general case you do:

```      
for (unsigned int i=0; i< weights.size(); ++i)
{
     averaged_parameter += weights[i]*std::pow(values[i],p);
}
return std::pow(averaged_parameter,1/p);
```
But if you want to do for example an arithmetic average (p=1) it is I think faster to do 
```
for (unsigned int i=0; i< weights.size(); ++i)
     averaged_parameter += weights[i]*values[i];
return averaged_parameter;
```
The result won't matter, but you don't have to  do weights.size() +1 times a power function. So that is also the case for the other special cases. So since the user of the function doesn't need to know how it's exactly implmented, I though it would be better not to put that in the documentation above the function, but only in the document with comments in the function what I am doing there.